### ISSUE TITLE
Sample rate conversion with SDL_AudioStream and WASAPI does not seem to work in 2.0.16

### ISSUE THREAD

**Issue by leonstyhre** [libsdl-org/SDL#4946]
Hi!
I just tried to use SDL 2.0.16 in in my project and on Windows with WASAPI the sample rate conversion no longer works.
I use SDL_AudioStream, SDL_AudioStreamPut, SDL_AudioStreamGet etc. to process audio and the issue can be reproduced by playing 48000 Hz sound and setting the output frequency to 44100 Hz in SDL. The audio then plays too slowly, but changing the output frequency to match the source audio frequency makes sound play at the correct speed.
The problem occurs both in Windows 10 and 8.1.

Replacing the 2.0.16 SDL2.dll file in my application folder with the one from 2.0.14 fixes the problem. Recompiling my application when changing the files makes no difference.

I've also tested my application on macOS with 2.0.16 and the problem does not occur there so I'm pretty sure it's a Windows problem, and possibly a WASAPI-specific issue.

Here's some code if you want to understand how I use the API:
https://gitlab.com/leonstyhre/emulationstation-de/-/blob/master/es-core/src/AudioManager.cpp

Many thanks in advance!

**Comment by slouken** [comment]
Are you seeing this in the latest snapshot?
http://www.libsdl.org/tmp/SDL-2.0.zip

**Comment by leonstyhre** [comment]
Thanks for your really fast reply! Yes that version worked fine, the audio now plays correctly!

I use the audio conversion for a video player in my application, and with the snapshot build it hangs for 1,5 seconds or so every time I stop a video, but I guess that could be because it's a debug build of the library? (Never seen this behavior in other SDL versions on any operating systems.)

Anyway, I will continue to use 2.0.14 for now and wait for the next stable release where I guess you will have this WASAPI fix included?

**Comment by slouken** [comment]
It's not a debug build, can you report the audio hang as a separate bug along with a stack trace and any other diagnostic info you can provide?

Thanks for confirming the WASAPI fix!

**Comment by leonstyhre** [comment]
The audio hang issue is a false alarm as far as SDL is concerned as amazingly it seems to be caused by a compiler bug. I switched from SDL 2.0.14 to 2.0.16 at the same time that I moved from MinGW 9.2.0 to 10.3.0 and with this compiler the execution time of the FFmpeg function avfilter_graph_free() went from a few milliseconds to hundreds of milliseconds or even seconds!

Recompiling with GCC 9.2.0 or MSVC completely eliminates the problem.

A good reminder to never change two things at the same time :)
